### ISSUE TITLE
Feature to pin most used blocks in the toolbar

### ISSUE THREAD

**Issue by callaloo** [WordPress/gutenberg#22402]
I am not sure if this question has been asked before, but is it possible for the team to introduce a feature that allows pinning some mostly used blocks in the toolbar without having to keep on pressing the + or the 'change block type" icon? For example, I use the heading block and the list block a lot and would appreciate a way to have them pinned in the toolbar to make sure they are one click away.

**Comment by mapk** [comment]
I'm hesitant to add more icons to the topbar. If you're looking for a way to quickly add a block to the page without having to click the `+`, I recommend using the `/` inserter. Just type `/` and the block name to add a new block to your document. Is this a possible solution for you?

**Comment by askel45** [comment]
> Is this a possible solution for you?

I normally convert most of my blocks as I type into other types, coz some of my posts involve copying data from PDF documents and editing it or rewriting as I go. So, having the blocks pinned would be a good solution. I normally keep clicking the q like the symbol to convert them, which is unintuitive sometimes for long articles.

**Comment by enriquesanchez** [comment]
I agree that showing the most used blocks on the toolbar will not scale well with the many constraints we have there, like small viewports or the top toolbar setting.

I, however, see how this might be a missing feature in the new inserter and can perhaps be added back again as a category?

![gutenberg-36-inserter](https://user-images.githubusercontent.com/3276087/90297376-dc6cef80-de53-11ea-8cd9-54bb4c690d33.png)

**Comment by askel45** [comment]
> I agree that showing the most used blocks on the toolbar will not scale well with the many constraints we have there, like small viewports or the top toolbar setting.
> 
> I, however, see how this might be a missing feature in the new inserter and can perhaps be added back again as a category?
> 
> ![gutenberg-36-inserter](https://user-images.githubusercontent.com/3276087/90297376-dc6cef80-de53-11ea-8cd9-54bb4c690d33.png)

This was more about pinning to the toolbar rather than using the most used category.

**Comment by enriquesanchez** [comment]
@askel45 Got it. I'm reverting the title of the issue to reflect your initial request.

**Comment by Mamaduka** [comment]
Closing as "maybe later". Why?

* The editor now has a "Show most used blocks" option. Displaying the most used blocks at the top of the Inserter.
* The top toolbar has very constrained space.
* The issue has been inactive for the last couple of years.

## Screenshot

<img width="770" height="702" alt="Image" src="https://github.com/user-attachments/assets/1e9bb45c-c369-4935-89d1-35b3a732ded3" />
